app/controllers/benefits/CompanyBenefitController.scala

```diff
@@ -113,20 +118,41 @@ class CompanyBenefitController @Inject()(
         }
       },
       success => {
-        success match {
-          case Some(NoIDontGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, NoIDontGetThisBenefit) map { _ =>
-              Redirect(controllers.benefits.routes.RemoveCompanyBenefitController.stopDate())
-            }
-          case Some(YesIGetThisBenefit) =>
-            journeyCacheService.cache(DecisionChoice, YesIGetThisBenefit) map { _ =>
-              Redirect(
-                controllers.routes.ExternalServiceRedirectController
-                  .auditInvalidateCacheAndRedirectService(TaiConstants.CompanyBenefitsIform)
-                  .url)
+        val benefitType = journeyCacheService.mandatoryJourneyValueAs(
+          EndCompanyBenefit_BenefitTypeKey,
+          x => x
+        )
+
+        benefitType.flatMap {
+          case Right(name) =>
+            success match {
+              case Some(NoIDontGetThisBenefit) =>
+                journeyCacheService
+                  .cache(getBenefitDecisionKey(Some(name), DecisionChoice), NoIDontGetThisBenefit)
+                  .map { _ =>
+                    Redirect(controllers.benefits.routes.RemoveCompanyBenefitController.stopDate())
+                  }
+              case Some(YesIGetThisBenefit) =>
+                journeyCacheService.cache(getBenefitDecisionKey(Some(name), DecisionChoice), YesIGetThisBenefit).map {
+                  _ =>
+                    Redirect(controllers.routes.ExternalServiceRedirectController
+                      .auditInvalidateCacheAndRedirectService(TaiConstants.CompanyBenefitsIform))
+                }
+              case e => {
+                logger.error(s"Data in incorrect format: $e")
+                Future.successful(Redirect(controllers.routes.TaxAccountSummaryController.onPageLoad()))
+              }
             }
+
+          case Left(e) => {
+            logger.error(s"Data not found in cache: $e")
```

Comment:
Don't need this logger as it is logged at the connector level

Reply:
Removed